File: Quick Sort/quicksort.c
#include <stdio.h>

int partition(int v[], int p, int r){
    int x = v[r];
    int i = p - 1;
    for(int j = p; j <= r - 1; j++){
        if(v[j] <= x){
            i++;
            int temp = v[i];
            v[i] = v[j];
            v[j] = temp;
        }
    }
    int temp = v[i + 1];
    v[i + 1] = v[r];
    v[r] = temp;
    return i + 1;
}

void quicksort(int v[], int p, int r){
    if(p < r){
        int q = partition(v, p, r);
        quicksort(v, p, q - 1);
        quicksort(v, q + 1, r);
    }
}

int main() {
    int arr[] = { 10, 7, 8, 9, 1, 5 };
    int n = sizeof(arr)/sizeof(arr[0]);
    quicksort(arr, 0, n - 1);
    for(int i = 0; i < n; i++){
        printf("%d ", arr[i]);
    }
    printf("\n");
}

File: Quick Sort/quicksort.py
def quicksort(v,p,r):
    if p < r:
        q = partition(v,p,r)
        quicksort(v,p,q - 1)
        quicksort(v,q + 1, r)
    return v
        
def partition(v,p,r):
    x = v[r]
    i = p - 1
    for j in range(p, r):
        if v[j] <= x:
            i = i + 1
            v[i], v[j] = v[j], v[i]
    v[i + 1], v[r] = v[r], v[i + 1]
    return i + 1

def main():
    v = [6,2,5,1,4,3]
    print(quicksort(v, 0 , len(v) - 1))

if __name__ == "__main__":
    main()

